### PR TITLE
Add Digital Ocean Cloud Instructions for Kubernetes

### PR DESCRIPTION
**Pull request by alexmorley**
Digital Ocean now has a (beta) kubernetes offering.

I am adding some guidelines for the step-zero section of the documentation.

### CODE REVIEW

**Comment by willingc** [comment]
Thanks @alexmorley for the PR and the feedback. I believe that K8 on DO is now in general release: https://blog.digitalocean.com/digitalocean-releases-k8s-as-a-service/

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. _digital-ocean:
+
+Step Zero: Kubernetes on Digital Ocean
+----------------------------------------------------------------
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]: rST headings should match the length of the underscores.

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+You can create a Kubernetes cluster `either through the Digital Ocean website, or using the Digital Ocean command line tools <https://www.digitalocean.com/>`_.
+
+This page describes the commands required to setup a Kubernetes cluster using the command line.
+If you prefer to use the Digital Ocean portal see the `Digital Ocean Get Started <https://www.digitalocean.com/products/kubernetes>`_.
+
+
+#. Prepare your Digital Ocean shell environment.
+   
+   * **Install command-line tools locally**. You'll need at least v1.13.0 to be able to new the kubernetes control.
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]:
```suggestion
   * **Install command-line tools locally**. You'll need at least v1.13.0.
```

Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]:
```suggestion
   a. **Install command-line tools locally**. You'll need at least v1.13.0 to be able to new the kubernetes control. 
```

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        sudo mv doctl /usr/bin/
+
+     Then run the following command to connect your local
+     CLI with your account:
+
+     .. code-block:: bash
+
+        doctl auth init
+
+     Before you do so you'll need to create an API token on the Digital Ocean portal. Navigate to API then Generate New Token.
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]:
```suggestion
     b. Create an API token on the Digital Ocean portal. Navigate to API then Generate New Token.
```

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+     You can either follow the `installation instructions <https://github.com/digitalocean/doctl/blob/master/README.md>` or use the commands below:
+
+     .. code-block:: bash
+
+        wget https://github.com/digitalocean/doctl/releases/download/v1.13.0/doctl-1.13.0-linux-amd64.tar.gz
+        tar -xvf doctl-1.13.0-linux-amd64.tar.gz
+        sudo mv doctl /usr/bin/
+
+     Then run the following command to connect your local
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]:
```suggestion
     c. Connect your local
```

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+     CLI with your account:
+
+     .. code-block:: bash
+
+        doctl auth init
+
+     Before you do so you'll need to create an API token on the Digital Ocean portal. Navigate to API then Generate New Token.
+
+#. Create your cluster.
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]: Let's verify that the beta environment variable is needed now.

Comment (alexmorley) [reply doc/source/digital-ocean/step-zero-digital-ocean.rst]: It was last week. Even though the service is out of beta I think the `doctl` support is still not. I will double check though.

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        sudo mv doctl /usr/bin/
+
+     c. Connect your local
+     CLI with your account:
+
+     .. code-block:: bash
+
+        doctl auth init
+
+     b. Create an API token on the Digital Ocean portal. Navigate to API then Generate New Token.
```
Comment (consideRatio) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]: b. follows c. typo

Comment (willingc) [reply doc/source/digital-ocean/step-zero-digital-ocean.rst]: Let's put the token creation before the auth init.

File: doc/source/digital-ocean/step-zero-digital-ocean.rst (hunk 5 above)
Comment (alexmorley) [reply doc/source/digital-ocean/step-zero-digital-ocean.rst]: Yep. Still in beta on master (and the latest release). https://github.com/digitalocean/doctl/blame/1bc1c5ec221e0492848fca2fdae124edcd9c7c8b/commands/kubernetes.go#L86

**Comment by alexmorley** [comment]
> Wieeeeee!!! Thanks for leading the way on this @alexmorley! Looks good to me but i have not tried following the instructions myself.

Yes If anyone can try this on their DO it would be great to have independent checks before merging?

> Im curious to learn about if they (managed kubernetes on Digital ocean) provide a managed cluster autoscaler that adds/removes nodes based on pods pending / nodes being empty

Apparently it will be released this year (https://www.digitalocean.com/community/questions/will-kubernetes-have-autoscaling-by-default).

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+This page describes the commands required to setup a Kubernetes cluster using the command line.
+If you prefer to use the Digital Ocean portal see the `Digital Ocean Get Started <https://www.digitalocean.com/products/kubernetes>`_.
+
+
+#. Prepare your Digital Ocean shell environment.
+   
+   a. **Install command-line tools locally**. You'll need at least v1.13.0. 
+
+     You can either follow the `installation instructions <https://github.com/digitalocean/doctl/blob/master/README.md>` or use the commands below:
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]: Let's correct the indentation for the a/b/c. See `View file` for a better view of this.

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+     b. Create an API token on the Digital Ocean portal. Navigate to API then Generate New Token.
+
+     c. Connect your local CLI with your account:
+
+     .. code-block:: bash
+
+        doctl auth init
+
+#. Create your cluster.
+   Digital Ocean's kubernetes support is in beta so you'll need to run the following (add it to your .bashrc if you want to make this change permenant).
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]:
```suggestion
   Digital Ocean's use of doctl for kubernetes support is in beta so you'll need to run the following (add it to your ``.bashrc`` if you want to make this change permanent).
```

File: doc/source/digital-ocean/step-zero-digital-ocean.rst [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+   .. code-block:: bash
+
+      kubectl get node
+
+   The response should list three running nodes and their Kubernetes versions!
+   Each node should have the status of ``Ready``, note that this may take a
+   few moments.
+
+.. note::
```
Comment (willingc) [inline doc/source/digital-ocean/step-zero-digital-ocean.rst]:
```suggestion
```

**Comment by minrk** [comment]
Thanks!

**Comment by consideRatio** [comment]
Thanks :tada: :heart:!

**Pull request merged**